### ISSUE TITLE
Screen Reader: Many Pages: The search field is announced without a role

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77535]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS, open the app
2. Swipe right to focus the Search button and double-tap to activate it
3. Swipe right to focus the Search field and note how it is announced

This is a global issue affecting all screens with the Search field
Examples:
1. Group Chat: Start Chat
2. Profile: Country
3. Profile: Timezone
4. Wallet: Add Bank Account
5. Create Expense: Merchant
6. Reports
7. Create Workspace > Company Cards: Add Card

## Expected Result:
The field should be announced with a search field role, e.g.:
- 'Search for something, search field'

## Actual Result:
The search field is currently announced only with its label and is missing its programmatic role:
- 'Search for something'

## Area issue was found in:
Search

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Ensure the search field is implemented using a proper TextInput element. Do not wrap the TextInput in a container that overrides or blocks the announcement of the role.
https://appt.org/en/docs/react-native/samples/accessibility-role
https://reactnative.dev/docs/textinput

## Environment:
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)

## Screenshots/Videos

<img width="1179" height="2443" alt="Image" src="https://github.com/user-attachments/assets/9f9a8c0a-8fca-4202-a315-c6133a4a55d8" />

https://github.com/user-attachments/assets/8762b43b-6a17-4161-bf32-a2c7522d1fe2

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7009048
**Issue reported by:** NA
**Slack conversation:** NA
